planet/src/main/webapp/places.js

```diff
@@ -198,10 +221,10 @@ function callback(place, status) {
             placeDetails['Website'] = place.url;
         }
         placeInfo.push(placeDetails);
-    }    
+    }
     // List results when all callbacks are finished
     count ++ 
-    if (count === places.length) {
+    if (count === 10) {
         count = 0; 
         listResults();
```

Comment:
doesn't listresults clear the currently displayed results? If so, the second set of requests will replace the first.

I'd suggest taking a step back and thinking holistically about what the sequence of events should be. It might lead you to a cleaner solution to this problem. Happy to help with this design process if that's helpful